pkg/virtual/apiexport/authorizer/binding_test.go

```diff
@@ -0,0 +1,517 @@
+				return &apisv1alpha2.APIBinding{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "bar",
+					},
+					Spec: apisv1alpha2.APIBindingSpec{
+						PermissionClaims: []apisv1alpha2.AcceptablePermissionClaim{
+							{
+								PermissionClaim: apisv1alpha2.PermissionClaim{
+									GroupResource: apisv1alpha2.GroupResource{
+										Group:    "foo",
+										Resource: "bar",
+									},
+									All:   true,
+									Verbs: []string{"get"},
+								},
+								State: apisv1alpha2.ClaimAccepted,
+							},
+						},
+					},
+				}, nil
+			},
+			getAPIExport: func(clusterName, apiExportName string) (*apisv1alpha2.APIExport, error) {
+				return &apisv1alpha2.APIExport{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "bar",
+					},
+					Spec: apisv1alpha2.APIExportSpec{
+						PermissionClaims: []apisv1alpha2.PermissionClaim{
+							{
+								GroupResource: apisv1alpha2.GroupResource{
+									Group:    "foo",
+									Resource: "bar",
+								},
+								All:   true,
+								Verbs: []string{"get", "list"},
+							},
+						},
+					},
+				}, nil
+			},
+			expectedDecision: authorizer.DecisionDeny,
+			expectedReason:   "failed to find suitable reason to allow access in APIBinding",
+		},
+		{
+			name: "list request to the API bound via permission claims but not allowing list (3)",
+			attr: &authorizer.AttributesRecord{
+				User:     &user.DefaultInfo{},
+				APIGroup: "foo",
+				Resource: "bar",
+				Verb:     "list",
+			},
+			apidomainKey: apidomainKey,
+			getAPIBindingByExport: func(clusterName, apiExportName, apiExportCluster string) (*apisv1alpha2.APIBinding, error) {
+				return &apisv1alpha2.APIBinding{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "bar",
+					},
+					Spec: apisv1alpha2.APIBindingSpec{
+						PermissionClaims: []apisv1alpha2.AcceptablePermissionClaim{
+							{
+								PermissionClaim: apisv1alpha2.PermissionClaim{
+									GroupResource: apisv1alpha2.GroupResource{
+										Group:    "foo",
+										Resource: "bar",
+									},
+									All:   true,
+									Verbs: []string{"get", "list"},
+								},
+								State: apisv1alpha2.ClaimAccepted,
+							},
+						},
+					},
+				}, nil
+			},
+			getAPIExport: func(clusterName, apiExportName string) (*apisv1alpha2.APIExport, error) {
+				return &apisv1alpha2.APIExport{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "bar",
+					},
+					Spec: apisv1alpha2.APIExportSpec{
+						PermissionClaims: []apisv1alpha2.PermissionClaim{
+							{
+								GroupResource: apisv1alpha2.GroupResource{
+									Group:    "foo",
+									Resource: "bar",
+								},
+								All:   true,
+								Verbs: []string{"get"},
+							},
+						},
+					},
+				}, nil
+			},
+			expectedDecision: authorizer.DecisionDeny,
+			expectedReason:   "failed to find suitable reason to allow access in APIBinding",
+		},
+	} {
+		t.Run(tc.name, func(t *testing.T) {
```

Comment:
I think we are missing tests for:
1. I give/consent more verbs/permissions than apiexport needs - should allow
2. custom verbs like "connect" or "proxy".

Reply:
If the APIExport author removes a permission, that permission is no longer allowed even though it's still allowed via APIBinding. Only permissions that are allowed both in APIExport and APIBinding are allowed/permitted. However, I don't see (in the ideal scenario) how removing some permissions in APIExport might break things down. Sure, if they remove permissions that are still actually used, but then don't remove those permissions, I think that's the behavior that we want to encourage here.

Reply:
> I don't see (in the ideal scenario) how removing some permissions in APIExport might break things down.
this is what I want to prove in the test. That having more permissions on APIBinding side than its actually still works. 

Other way around - yes, it will break. But I assume case like this should work.

Reply:
I think we do have tests for these scenarios in this file (`pkg/virtual/apiexport/authorizer/binding_test.go`):

- `list request when APIExport allows list but APIBinding does not`
- `list request when APIBinding allows list but APIExport does not`

Have you checked these two cases? Both are resulting in a failure and that's correct.

> But I assume case like this should work.

I'm not sure what do you exactly mean by that, can you please clarify?